Less API!

bvanderveen opened the pull request:
Overall goal was to generally cut down on public the API of the Gate.dll project in preparation for a 1.0 release.
- All thread-aware code moved into Gate.Helpers (InputStream, OutputStream, Spool, Signal, Request, Response class). 
- Environment class now extends from IDictionary<string, object>. Host handlers could allocate it once (like they would with the dict) and pass it on. middlewares which want to use an Environment instance do a simple check like: var env = env as Environment ?? new Environment(envDict). previously a new Environment object had to be created to wrap the context at every middleware that wants to use it—now the stack is spared these extra allocations.
- AppBuilder/AppDelegate vs. AppBuilderExt/AppAction—needed to pick one and stick with it, and AppBuilderExt/AppAction got the axe. The ToDelegate method in Delegates.cs is nice enough that you can just use that if you need to work with an AppAction. And anyway, this was really only a work-around for a compiler bug and should work transparently in C# 5: http://stackoverflow.com/questions/4466859/delegate-system-action-does-not-take-0-arguments-is-this-a-c-compiler-bug/4467128#4467128
- Refactored the configuration loader to better expose its string-parsing bits—needed it for the upcoming Kayak CLI. :)
- Refactored AppBuilder to cut down on public API. Now there's an IAppBuilder interface which just exposes Use and Build. All other methods (Run, Map, Cascade) are implemented as extensions on top of this.
- All public API of Gate.dll is in the "Gate" namespace.

bvanderveen committed:
Env extends from dict

bvanderveen committed:
factored configuration loader to expose string -> (type, method) resolution

bvanderveen committed:
fix tests broken Environment change

bvanderveen committed:
Moved all thread-aware code into Gate.Helpers

bvanderveen committed:
simplify AppBuilder interface a bit

bvanderveen committed:
refactored AppBuilder for even less public API. introduced IAppBuilder to boil it down to the minimum necessary interface (Use and Build). Run and Map are now implemented as extensions. should be mostly source-compatible with the old way.

bvanderveen committed:
small cleanup of extensions

bvanderveen committed:
removed the Ext stuff. It's too easy for client code to just use Delegates.ToDelegate to justify the extra API within Gate.

bvanderveen committed:
DefaultConfigurationLoader -> GateConfigurationLoader, remove/sort usings

bvanderveen committed:
yikes, broke the build without realizing it. VS caching output or something…

bvanderveen committed:
namespaced everything to just Gate

bvanderveen committed:
hide DotByDot

jasonsirota commented:
which commit has the AppBuilder/AppDelegate vs. AppBuilderExt/AppAction change?

bvanderveen commented:
This one: https://github.com/bvanderveen/gate/commit/d9434dc761ba56cb68f605d766146c6b88d552ee

Just a deletion.

bvanderveen committed:
ConfigurationLoader based on Ragel.
Just wanted to get this in here because I beat my head against it for so long; ultimately though it's not the right approach I think.

bvanderveen committed:
and remove…

bvanderveen committed:
AppBuilder.BuildFromConfiguration -> BuildConfiguration, GateConfigurationLoader -> ConfigurationLoader

bvanderveen committed:
rename files based on class name

bvanderveen committed:
Added Kayak Host

bvanderveen committed:
Merge branch 'master' of ../gate

bvanderveen commented:
This pull is far too hairy. Going to break it down.
